### DISCUSSION TITLE
Support for Youtube Music as music provider

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#433] [Feature requests and ideas]
Not even sure if it's possible but can have a look.

**Comment by Symbiot78** [comment]
add option to search for artist, track, album etc.
start artist radio
shuffle selection (select artistor album -  shuffle whatever the selected result(s) are).
See and interact with liked artists, tracks, albums
see and interact with subscribed artists
Play playlists

**Comment by rjgrandy** [comment]
I second this! This would be fantastic.

**Comment by nevir** [comment]
Definitely interested in this as well.

Take a look at https://github.com/KoljaWindeler/ytube_music_player for some integration inspiration as well

**Comment by caj380** [comment]
I am also very interested in this! I currently use yTube Music Player in Home Assistant, but it is nowhere near as nice as Music Assistant! I regret cancelling my Spotify last month.

**Comment by LukaszP2** [comment]
how we can help you with that?

**Comment by Nemesis24** [comment]
I'm interested too !!

**Comment by marcelveldt** [comment]
Please show your interest for a feature request with the "like" button (thumbs up) on the original/top message, thanks!

**Comment by MarvinSchenkel** [comment]
Hey @marcelveldt , any way we can help you out with this one?

**Comment by sparkydave1981** [comment]
I'm very keen to see my DLNA server files accessible in MA

**Comment by marcelveldt** [comment]
> Hey @marcelveldt , any way we can help you out with this one?

Not yet, unless you want to help writing a music provider plugin for Music Assistant. In that case, contact me by PM in discord and I can share some development info. It's all written by a pluggable architecture but its lacking documentation yet :-)

Otherwise I think this i something I can pickup next week considering the enormous amount of votes

**Comment by sparkydave1981** [comment]
Sorry, commented on wrong issue

**Comment by MarvinSchenkel** [comment]
> > Hey @marcelveldt , any way we can help you out with this one?
> 
> Not yet, unless you want to help writing a music provider plugin for Music Assistant. In that case, contact me by PM in discord and I can share some development info. It's all written by a pluggable architecture but its lacking documentation yet :-)
> 
> Otherwise I think this i something I can pickup next week considering the enormous amount of votes

I unfortunately don't have the time to write a music provider from scratch (something with a 4-month old kid ;-)). Would love to do beta-testing when you get a first version out. I could also help add new features when there's a first version of the provider :-).

**Reply by AquaWolf** [reply]
Found an implementation here https://github.com/sigma67/ytmusicapi
It actually just supports a json file with auth header information so I think the session will not last that long.

**Reply by marcelveldt** [reply]
Thanks I will take a look at that once I start working on this. Based on the number of votes I think I really should have a try on making this work ;-)

**Comment by emeyedeejay** [comment]
Another vote for this - would be amazing to have access to my youtube music library! Music Assistant is AWESOME!

**Comment by wjbeckett** [comment]
Yet another vote for this.
Once YouTube Music is available as a music source I'll be able to use Music Assistant for everything!

**Comment by marcelveldt** [comment]
The first beta of Music Assistant with Youtube Music has just been released!

https://github.com/music-assistant/hass-music-assistant/releases/tag/2022.7.0b1

**Reply by wjbeckett** [reply]
Amazing! Trying it out now!

**Reply by wjbeckett** [reply]
For those wondering, You need the header cookie.
You can get that here:
https://ytmusicapi.readthedocs.io/en/latest/setup.html#copy-authentication-headers

Just grab the "cookie".

**Comment by enrique** [comment]
You can see how get it here:

https://github.com/music-assistant/hass-music-assistant/discussions/606

Enrique

**Comment by marcelveldt** [comment]
Now that this feature has been implemented I've locked the discussion as it is no longer a feature request but ongoing business ;-)

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
